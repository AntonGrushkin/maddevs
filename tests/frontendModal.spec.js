import {
  mount
} from '@vue/test-utils';
import frontendModal from '@/components/Modals/frontend-modal';

describe('Frontend modal', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(frontendModal, {
      stubs: ['ValidationProvider', 'ValidationObserver', 'modal']
    });
  });

  // ------ IMPORTANT ----- //
  test('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy();
  });
  
  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
  // --------------------- //

  test('sets the correct default data', () => {
    expect(typeof frontendModal.data).toBe('function');
    const defaultData = frontendModal.data();
    expect(
      defaultData.agreeWithPrivacyPolicy &&
      defaultData.agreeToGetMadDevsDiscountOffers
    ).toEqual(false);
    expect(defaultData.inputId).toEqual('frontend');
  });

  test('has a functions', () => {
    expect(
      typeof frontendModal.methods.getPrivacyCheckboxState && 
      typeof frontendModal.methods.getDiscountOffersCheckboxState && 
      typeof frontendModal.methods.autosize
    ).toBe('function');
  });

  test('call functions with params and change variables state', () => {
    wrapper.vm.getPrivacyCheckboxState(true);
    wrapper.vm.getDiscountOffersCheckboxState(true);

    expect(
      wrapper.vm.$data.agreeWithPrivacyPolicy &&
      wrapper.vm.$data.agreeToGetMadDevsDiscountOffers
    ).toEqual(true);
  });

  test('autosize function should add value in event key', () => {
    const event = {
      target: {
        style: {
          height: ''
        },
        scrollHeight: 100
      }
    }

    wrapper.vm.autosize(event);
    expect(event.target.style.height).toEqual('100px');
  });
});
