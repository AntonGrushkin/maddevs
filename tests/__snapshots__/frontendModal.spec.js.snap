// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Frontend modal renders correctly 1`] = `
<modal-stub
  name="frontend"
>
  <img
    alt="Close modal"
    class="close-modal"
    src="../../assets/img/common/close-icon.svg"
  />
   
  <validationobserver-stub />
</modal-stub>
`;
